ambry-account/src/main/java/com/github/ambry/account/CompositeAccountService.java

```diff
@@ -40,20 +41,20 @@
   private static final String ACCOUNT_DATA_CONSISTENCY_CHECKER_PREFIX = "account-data-consistency-checker";
   private final AccountService primaryAccountService;
   private final AccountService secondaryAccountService;
-  private final AccountServiceMetrics accountServiceMetrics;
   private final CompositeAccountServiceConfig config;
   private final ScheduledExecutorService scheduler;
   private static final Random random = new Random();
+  private AtomicInteger accountsMismatchCount = new AtomicInteger(0);
```

Comment:
Minor question: do we really need to keep track of accurate mismatch count?  I see this value increases only and never decrease (It's a Gauge type in the metrics). I think maybe an error counter is sufficient. I think there is transient cases where two AS are inconsistent but they may quickly converge.

Reply:
It should be reset to 0 before each compare is made.  If it isn't that needs to be added.

Reply:
The value is set in this line:
`accountsMismatchCount.set(
        AccountUtils.compareAccounts(primaryAccountService.getAllAccounts(), secondaryAccountService.getAllAccounts())); `

But there are other places where it gets incremented that don't make sense for a guage.